src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -117,7 +124,7 @@ public String apply(Node input) {
    * Aggregates all emitted types, used in a final pass to find types emitted in type position but
    * not declared, possibly due to missing goog.provides.
    */
-  private final Set<String> typesUsed = new LinkedHashSet<>();
+  private final Set<String> typesUsed = new TreeSet<>();
```

Comment:
Adding fixed iteration order to make our tests run on JDK7 and 8 (see comment below). Turns out this was not the offending set, but I still see no reason not to use TreeSet everywhere.